raven/utilities/mk_test.py

```diff
@@ -0,0 +1,202 @@
+            tp[i] = sum(x == unique_x[i])
+        var_s = (n*(n-1)*(2*n+5) - np.sum(tp*(tp-1)*(2*tp+5)))/18
+
+    if s > 0:
+        z = (s - 1)/np.sqrt(var_s)
+    elif s < 0:
+        z = (s + 1)/np.sqrt(var_s)
+    else:  # s == 0:
+        z = 0
+
+    # calculate the p_value
+    p = 2*(1-norm.cdf(abs(z)))  # two tail test
+    h = abs(z) > norm.ppf(1-alpha/2)
+
+    if (z < 0) and h:
+        trend = 'decreasing'
+    elif (z > 0) and h:
+        trend = 'increasing'
+    else:
+        trend = 'no trend'
+
+    return trend, h, p, z
+
+
+def check_num_samples(beta, delta, std_dev, alpha=0.05, n=4, num_iter=1000,
+                      tol=1e-6, num_cycles=10000, m=5):
+    """
+    This function is an implementation of the "Calculation of Number of Samples
+    Required to Detect a Trend" section written by Sat Kumar Tomer
+    (satkumartomer@gmail.com) which can be found at:
+    http://vsp.pnnl.gov/help/Vsample/Design_Trend_Mann_Kendall.htm
+    As stated on the webpage in the URL above the method uses a Monte-Carlo
+    simulation to determine the required number of points in time, n, to take a
+    measurement in order to detect a linear trend for specified small
+    probabilities that the MK test will make decision errors. If a non-linear
+    trend is actually present, then the value of n computed by VSP is only an
+    approximation to the correct n. If non-detects are expected in the
+    resulting data, then the value of n computed by VSP is only an
+    approximation to the correct n, and this approximation will tend to be less
+    accurate as the number of non-detects increases.
+    Input:
+        beta: probability of falsely accepting the null hypothesis
+        delta: change per sample period, i.e., the change that occurs between
+               two adjacent sampling times
+        std_dev: standard deviation of the sample points.
+        alpha: significance level (0.05 default)
+        n: initial number of sample points (4 default).
+        num_iter: number of iterations of the Monte-Carlo simulation (1000
+                  default).
+        tol: tolerance level to decide if the predicted probability is close
+             enough to the required statistical power value (1e-6 default).
+        num_cycles: Total number of cycles of the simulation. This is to ensure
+                    that the simulation does finish regardless of convergence
+                    or not (10000 default).
+        m: if the tolerance is too small then the simulation could continue to
+           cycle through the same sample numbers over and over. This parameter
+           determines how many cycles to look back. If the same number of
+           samples was been determined m cycles ago then the simulation will
+           stop.
+        Examples
+        --------
+          >>> num_samples = check_num_samples(0.2, 1, 0.1)
+    """
+    # Initialize the parameters
+    power = 1.0 - beta
+    P_d = 0.0
```

Comment:
Variables shouldn't be composed of capital letters. The PEP8 gods will not be pleased.  
```suggestion
    p_d = 0.0
```

Reply:
Fixed!